### PR TITLE
Enumerables: Rspec tests

### PR DESCRIPTION
**Pull request by PhillipUg**
## In this branch, we have done the following;
  - We have added tests for all our enumerable methods
  - We have changed project file structure and added lib and spec folders
  - Followed other project-specific guidelines as required by microverse

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+    context 'when block given' do
+      let(:new_array1) { [] }
+      let(:new_array2) { [] }
+
+      it 'should call block with one argument, the item (element), for each collection item' do
+        array2.my_each { |ele| new_array1.push(ele + '!') }
+        range.my_each { |ele| new_array2.push(ele + 2) }
+        expect(new_array1).to eq(%w[two! three! ten! twelve!])
+        expect(new_array2).to eq([3, 4, 5, 6, 7])
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return an Enumerator ' do
+        expect(array1.my_each).to be_a_kind_of(Enumerator)
+        expect(array1.my_each.to_a).to eq(array1)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

Your second expect is not related to the `it` statement you have. With just the first expect you would be checking for the enumerator, so you don't need it. Check that on all your other tests.

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+      it 'should call block with two arguments, the item and its index for each collection item' do
+        array2.my_each_with_index { |ele, idx| hash[ele] = idx }
+        expect(hash).to eq({ 'ten' => 2, 'three' => 1, 'twelve' => 3, 'two' => 0 })
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return an Enumerator' do
+        expect(array1.my_each_with_index).to be_a_kind_of(Enumerator)
+        expect(array1.my_each_with_index.to_a).to eq(array1)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+      it 'should return an array of all collection items that evaluate to true' do
+        expect(range.my_select { |ele| ele > 2 }).to eq([3, 4, 5])
+        expect(array2.my_select { |ele| ele.size > 3 }).to eq(%w[three twelve])
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return an Enumerator' do
+        expect(array1.my_select).to be_a_kind_of(Enumerator)
+        expect(array1.my_select.to_a).to eq(array1)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        it 'should return false' do
+          expect(array3.my_all?).to be false
+        end
+      end
+    end
+
+    context 'when block given' do
+      it 'should return true if all of the collection items evaluate to true' do
+        expect(array5.my_all? { |ele| ele >= 1 }).to be true
+        expect(range.my_all? { |ele| ele.is_a?(Numeric) }).to be true
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+      end
+
+      it 'should return false if any of the collection items evaluates to false' do
+        expect(array3.my_all? { |ele| ele.is_a?(Integer) }).to be false
+        expect(range.my_all? { |ele| ele < 3 }).to be false
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+
+    context 'when pattern is given as argument' do
+      it 'should return true if pattern === each element in collection' do
+        expect(array1.my_all?(Integer)).to be true
+        expect(array2.my_all?(/t/)).to be true
+      end
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+
+      it 'should return false if pattern doesn\'t match each element in collection' do
+        expect(array3.my_all?(Integer)).to be false
+        expect(array1.my_all?(/e/)).to be false
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+        it 'should return false' do
+          expect(array4.my_any?).to be false
+        end
+      end
+    end
+
+    context 'when block given' do
+      it 'should return true if any of the collection items evaluates to true' do
+        expect(array5.my_any? { |ele| ele == 3 }).to be true
+        expect(range.my_any? { |ele| ele > 4 }).to be true
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+      end
+
+      it 'should return false if none of the collection items evaluates to true' do
+        expect(array2.my_any? { |ele| ele.size > 6 }).to be false
+        expect(range.my_any? { |ele| ele > 5 }).to be false
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+
+    context 'when pattern is given as argument' do
+      it 'should return true if pattern === any element in collection' do
+        expect(array3.my_any?(nil)).to be true
+        expect(array2.my_any?(/o/)).to be true
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+      end
+
+      it 'should return false if pattern doesn\'t match any element in collection' do
+        expect(array2.my_any?(Integer)).to be false
+        expect(array2.my_any?(/s/)).to be false
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 13]
```diff
@@ -0,0 +1,273 @@
...
+        it 'should return false' do
+          expect(array3.my_none?).to be false
+        end
+      end
+    end
+
+    context 'when block given' do
+      it 'should return true if none of the collection items evaluates to true' do
+        expect(array5.my_none? { |ele| ele == 4 }).to be true
+        expect(range.my_none? { |ele| ele > 5 }).to be true
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 14]
```diff
@@ -0,0 +1,273 @@
...
+      end
+
+      it 'should return false if any of the collection items evaluates to true' do
+        expect(array5.my_none? { |ele| ele == 3 }).to be false
+        expect(range.my_none? { |ele| ele > 4 }).to be false
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 15]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+
+    context 'when pattern is given as argument' do
+      it 'should return true if pattern doesn\'t match any element in collection' do
+        expect(array3.my_none?(String)).to be true
+        expect(array2.my_none?(/s/)).to be true
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 16]
```diff
@@ -0,0 +1,273 @@
...
+      end
+
+      it 'should return false if pattern === any element in collection' do
+        expect(array3.my_none?(nil)).to be false
+        expect(array2.my_none?(/o/)).to be false
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 17]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+  end
+
+  describe '#my_count' do
+    context 'when block given' do
+      it 'counts number of collection items yielding a true value' do
+        expect(array5.my_count { |ele| ele == 2 }).to eq(3)
+        expect(array1.my_count { |ele| ele > 2 }).to eq(2)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 18]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return number of items in collection' do
+        expect(array2.count).to eq(4)
+        expect(array4.count).to eq(4)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 19]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+  end
+
+  describe '#my_map' do
+    context 'when block given' do
+      it 'should return a new array with results of running block once for every collection item' do
+        expect(array1.my_map { |ele| ele + 1 }).to eq([2, 3, 4, 5])
+        expect(range.my_map { |ele| ele + 2 }).to eq([3, 4, 5, 6, 7])
+        expect(range.my_map { 'cat' }).to eq(%w[cat cat cat cat cat])
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 20]
```diff
@@ -0,0 +1,273 @@
...
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return an Enumerator ' do
+        expect(array1.my_map).to be_a_kind_of(Enumerator)
+        expect(array1.my_map.to_a).to eq(array1)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 21]
```diff
@@ -0,0 +1,273 @@
...
+  end
+
+  describe '#my_inject' do
+    context 'when block is given' do
+      context 'when no initial value is specified' do
+        it 'should combine all collection items by applying a binary operation' do
+          expect(range.my_inject { |acc, num| acc + num }).to eq(15)
+          expect(array1.my_inject { |acc, num| acc + num }).to eq(10)
+          expect(array1.my_inject { |acc, num| acc * num }).to eq(24)
+          expect(array2.my_inject { |acc, num| acc + num }).to eq('twothreetentwelve')
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 22]
```diff
@@ -0,0 +1,273 @@
...
+        end
+      end
+
+      context 'when initial value is specified' do
+        it 'should combine all collection items with binary operation starting with specified initial value' do
+          expect(range.my_inject(2) { |acc, num| acc + num }).to eq(17)
+          expect(array1.my_inject(2) { |acc, num| acc + num }).to eq(12)
+          expect(array1.my_inject(5) { |acc, num| acc * num }).to eq(120)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 23]
```diff
@@ -0,0 +1,273 @@
...
+        end
+      end
+    end
+
+    context 'when no block is given' do
+      context 'when only symbol is specified as argument' do
+        it 'should combine all collection items using symbol specified' do
+          expect(array1.my_inject(:+)).to eq(10)
+          expect(range.my_inject(:+)).to eq(15)
+          expect(array1.my_inject(:*)).to eq(24)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 24]
```diff
@@ -0,0 +1,273 @@
...
+        end
+      end
+
+      context 'when initial value and symbol are specified as arguments' do
+        it 'should combine all collection items begining with initial value using symbol specified' do
+          expect(array1.my_inject(2, :+)).to eq(12)
+          expect(range.my_inject(2, :+)).to eq(17)
+          expect(array1.my_inject(5, :*)).to eq(120)
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 25]
```diff
@@ -0,0 +1,273 @@
...
+      it 'should return false if pattern doesn\'t match each element in collection' do
+        expect(array3.my_all?(Integer)).to be false
+        expect(array1.my_all?(/e/)).to be false
+      end
+    end
+  end
+
+  describe '#my_any?' do
+    context 'when no block is given' do
+      context 'if  atleast one of elements is not (false) or (nil)' do
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: **Optional**: try to keep your nesting simpler. If you write:
```
describe '#my_any?' do
    context 'when no block is given' do
        it 'should return true 'if at least one of elements is not (false) or (nil)'' do
          expect(array3.my_any?).to be true
        end
     end
// other context...
end
```
The code is more readable that way, and you only keep two contexts: `when block is given` and `when no block is given`.

Keep this in mind for all the other tests, if you decide to implement the change.

File: spec/enumerables_spec.rb [hunk 26]
```diff
@@ -0,0 +1,273 @@
...
+
+  describe '#my_select' do
+    it 'should raise an error when an argument is given' do
+      expect { array1.my_select(1) }.to raise_error(ArgumentError)
+    end
+
+    context 'when block given' do
+      it 'should return an array of all collection items that evaluate to true' do
+        expect(range.my_select { |ele| ele > 2 }).to eq([3, 4, 5])
+        expect(array2.my_select { |ele| ele.size > 3 }).to eq(%w[three twelve])
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerables_spec.rb [hunk 27]
```diff
@@ -0,0 +1,273 @@
...
+      it 'should return an Enumerator' do
+        expect(array1.my_select).to be_a_kind_of(Enumerator)
+        expect(array1.my_select.to_a).to eq(array1)
+      end
+    end
+  end
+
+  describe '#my_all?' do
+    context 'when no block is given' do
+      context 'if none of elements is (false) or (nil)' do
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: **Optional**: try to keep your nesting simpler. If you write:
```
describe '#my_all?' do
    context 'when no block is given' do
        it 'should return true 'if none of elements is (false) or (nil)'' do
          expect(array3.my_all?).to be true
        end
     end
// other context...
end
```
The code is more readable that way, and you only keep two contexts: `when block is given` and `when no block is given`.

Keep this in mind for all the other tests, if you decide to implement the change.

File: spec/enumerables_spec.rb [hunk 28]
```diff
@@ -0,0 +1,273 @@
...
+
+  describe '#my_any?' do
+    context 'when no block is given' do
+      context 'if  atleast one of elements is not (false) or (nil)' do
+        it 'should return true' do
+          expect(array3.my_any?).to be true
+        end
+      end
+
+      context 'if none of the elements is not (false) or (nil)' do
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: It should be: if none of the elements is (false) or (nil)

File: spec/enumerables_spec.rb [hunk 29]
```diff
@@ -0,0 +1,273 @@
...
+      it 'should return false if pattern doesn\'t match any element in collection' do
+        expect(array2.my_any?(Integer)).to be false
+        expect(array2.my_any?(/s/)).to be false
+      end
+    end
+  end
+
+  describe '#my_none?' do
+    context 'when no block is given' do
+      context 'if none of the elements is not (false) or (nil)' do
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: Check the optional change about nested contexts suggested above.

File: spec/enumerables_spec.rb [hunk 30]
```diff
@@ -0,0 +1,273 @@
...
+      it 'should return an Enumerator ' do
+        expect(array1.my_map).to be_a_kind_of(Enumerator)
+        expect(array1.my_map.to_a).to eq(array1)
+      end
+    end
+  end
+
+  describe '#my_inject' do
+    context 'when block is given' do
+      context 'when no initial value is specified' do
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: Check the optional change about nested contexts suggested above.

File: spec/enumerables_spec.rb [hunk 31]
```diff
@@ -0,0 +1,273 @@
...
+        it 'should combine all collection items with binary operation starting with specified initial value' do
+          expect(range.my_inject(2) { |acc, num| acc + num }).to eq(17)
+          expect(array1.my_inject(2) { |acc, num| acc + num }).to eq(12)
+          expect(array1.my_inject(5) { |acc, num| acc * num }).to eq(120)
+        end
+      end
+    end
+
+    context 'when no block is given' do
+      context 'when only symbol is specified as argument' do
```
Comment (majovanilla) [inline spec/enumerables_spec.rb]: Check the optional change about nested contexts suggested above.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hey team, 👋

## You did a *great* job on this project! 💜 

- Your test coverage was excellent.

There are still some changes 👇 you need to implement to approve this request 🔍
- Some of your tests have typos, please take a look at them and correct them. 
- Check the comments below.


### Optional Changes:

If you want to make your project even better 🔥💥, I left a few suggestions you can implement. 
These are completely [Optional]: 🦔

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Comment by PhillipUg** [comment]
Thanks @majovanilla for your detailed review :smile: . Let us implement the suggested changes and make another request

**Comment by PhillipUg** [comment]
## We have corrected the following:
 - Implemented one expect statement for each 'it' block
 - resolved nesting complexity with the multiple 'context' blocks

**Review by Torres-ssf — APPROVED** [review]
### Project Approved :tada: :muscle: 
Hi there @PhillipUg :hand: 
Great job implementing the requested changes. Awesome job on this project :clap: 
**This PR approvoved**
Good luck with the next project

Happy coding :computer: 

**Sergio Torres**

**Comment by PhillipUg** [comment]
Thanks @Torres-ssf :smile:

**Pull request merged**